rfcs/20221124-monorepo.md

```diff
@@ -0,0 +1,446 @@
+
+1. The API is already organised by stack folders so the `frontend/` directory
+   will fit right in with the others like `api/` and `ingestion_server/`.
+   Similarly the scripts repo is nicely organised in folders, reducing
+   conflicts.
+
+1. The API and frontend share identical tooling for Git hooks, linting and
+   formatting. We will fight our tools less and encounter minimal friction.
+
+   - The frontend's approach for `pre-commit` expanded this type of usage to the
+     API as well!
+
+   - We're expanding the use of double-quoted strings to JavaScript to further
+     unify our style guides.
+
+1. The entire system can be integration tested during releases. The real API,
+   populated with test data, can replace the Talkback server as long as we
+   disable network calls and make output deterministic.
+
+The `WordPress/openverse` repo will absorb the `WordPress/openverse-api` and
+`WordPress/openverse-frontend` repos. The `WordPress/openverse-catalog` will
+also be merged, _later_.
+
+### Reference
+
+I'm following the steps listed below in a fork at
+<s>[@dhruvkb/monopenverse](https://github.com/dhruvkb/monopenverse/)</s>
+[@dhruvkb/monoverse](https://github.com/dhruvkb/monoverse/). You can refer to
+the fork, but note that it is a comes from a place of haste and has not been
+treated with the same level of love and care that the final treatment will
+receive.
+
+### Step 0: Prerequisites
+
+#### Get the timing right
+
+The first step will be to release the API and frontend, call a code freeze on
+both of them and pause work on both. This is to prevent the repos from
+continuing to drift as we merge a snapshot of them into the
+`WordPress/openverse` repo.
+
+This can prove difficult given how productive our team is, so we will need to
+channel this productivity towards the catalog in the meantime. I can foresee the
+end-to-end migration taking one week (ideal scenario) to becoming workable
+again, and another week (for us to iron out any gaps in the docs and
+references).
+
+##### Timeline breakdown
+
+- Day 1: Merging the repos and resolving conflicts, restoring broken workflows
+  except deploys
+- Day 2: Restoring deployment workflows including staging auto-deploy
+- Day 3: Transfer of issues from individual repos to monorepo
+- Day 4: Documentation fixes
+- Day 5: Housekeeping
+
+The second week is planned as a buffer in case any of these tasks ends up taking
+more time than a day, if something breaks or if someone falls ill etc. The ideal
+scenario is that we're completely back next week, the worst one takes two.
+
+Note that in the transition period nothing will break. The old repos will
+continue to exists as they are, till we ensure everything works and then we
+archive the current split repos.
+
+### Step 1: Merge with histories
+
+This should be quick save for a few merge conflicts. In case of conflict copy
+the code from [@dhruvkb/monoverse](https://github.com/dhruvkb/monoverse/).
```

Comment:
```suggestion
This should be quick save for a few merge conflicts. In case of conflict copy
the code from [@dhruvkb/monoverse](https://github.com/dhruvkb/monoverse/).
```

This instruction feels a little clear to the point of being potentially destructive. For example, if some code that was merge conflicted is waaay out of date in `monoverse`. Should this be edited to specify something like 'configuration files' instead of any code? Even that might be concerning, for example if we've updated an action version in the real repository that is behind in monoverse.

Or is monoverse meant to be kept up to date with the real repositories?

Reply:
Keeping `monoverse` (which was a testing ground and experimentation repo) updated witn the three repos being merged will be a huge investment of time and effort with no significant return. I've updated the instructions in 5763d16 to **not copy** the code but rather the resolution logic from the commits.